### ISSUE TITLE
ios_banner - IndexError

### ISSUE THREAD

**Issue by shepherdjay** [ansible-collections/cisco.ios#474]
##### SUMMARY

When running the ios_banner module a cisco ios-xe device returns a failure with a traceback. Another device reports fine status... devices are running same version and are same model.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

ios_banner

##### ANSIBLE VERSION

```paste below
ansible [core 2.12.1]
  config file = None
  configured module search path = ['/Users/jay.shepherd/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jay.shepherd/PycharmProjects/ansible_testing/.venv/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/jay.shepherd/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/jay.shepherd/PycharmProjects/ansible_testing/.venv/bin/ansible
  python version = 3.9.0 (v3.9.0:9cf6752276, Oct  5 2020, 11:29:23) [Clang 6.0 (clang-600.0.57)]
  jinja version = 3.0.3
  libyaml = True

```

##### COLLECTION VERSION

```paste below
Collection Version
---------- -------
cisco.ios  2.5.0 
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Cisco 9200-48p, 17.03.04b

##### STEPS TO REPRODUCE

From some testing it appears the problem may be due to the existing banner having been copied pasted from another device resulting in the config having multiple C's on the first line, ie:

```
banner login ^CCC
**** THIS IS A BANNER ***
^C
```

```yaml
- name: Update banner details
  hosts: all
  tasks:
    - name: Configure Login Banner
      cisco.ios.ios_banner:
        banner: login
        text: |
          *** EXAMPLE BANNER ***
        state: present
    - name: Remove the motd banner
      cisco.ios.ios_banner:
        banner: motd
        state: absent
```

##### EXPECTED RESULTS

Should complete

##### ACTUAL RESULTS

```paste below
The full traceback is:
Traceback (most recent call last):
  File "/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py", line 107, in <module>
    _ansiballz_main()
  File "/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_banner', init_globals=dict(_module_fqn='ansible_collections.cisco.ios.plugins.modules.ios_banner', _modlib_path=modlib_path),
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 210, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py", line 207, in <module>
  File "/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py", line 196, in main
  File "/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py", line 150, in map_config_to_obj
IndexError: list index out of range
fatal: [ommitted]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/Users/shepherdjay/.ansible/tmp/ansible-local-528094t6fwxan/ansible-tmp-1640040512.976873-52865-105766041660389/AnsiballZ_ios_banner.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_banner', init_globals=dict(_module_fqn='ansible_collections.cisco.ios.plugins.modules.ios_banner', _modlib_path=modlib_path),\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py\", line 207, in <module>\n  File \"/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py\", line 196, in main\n  File \"/var/folders/0l/kp389hfs0gxg1cdsr5wf6d9m0000gp/T/ansible_cisco.ios.ios_banner_payload_n7qxqdif/ansible_cisco.ios.ios_banner_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_banner.py\", line 150, in map_config_to_obj\nIndexError: list index out of range\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

```

**Comment by KB-perByte** [comment]
Hey @shepherdjay, I have added the similar config, but not able to reproduce the issue reported on XE device.
every run reports successful, can you try explicitly mentioning a `multiline_delimiter` that might be exclusive of the special/characters used in the banner value.
```
!         
banner login ^C
*** EXAMPLE BANNER ***
^C        
!         
```
and, is the issue consistent over multiple runs?

**Comment by shepherdjay** [comment]
@KB-perByte - I've been able to resolve this. The issue is the pip install version of ansible does not include the latest `cisco.ios` collection of 2.6.0 where it appears this issue was fixed. I had assumed I was running the latest version until I tested on a separate vm machine where I had done the install as apt-get and manual galaxy collection install of cisco.ios

running ansible-galaxy collection install cisco.ios resolved the issue

**Comment by shepherdjay** [comment]
Looks like this was the relevant fix #50 - I'll go ahead and close this issue. thanks

**Comment by KB-perByte** [comment]
hey @shepherdjay that is great there were some fixes that went in for ios_banner in the last release.
Closing this issue then.
Thanks.
